### PR TITLE
Add runtime engines to containerd ci workflow

### PR DESCRIPTION
**Pull request by hickeyma**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Adds runtimes to the containerd critest. Expands the test cases.

#### Which issue(s) this PR fixes:

Partial #613

#### Does this PR introduce a user-facing change?

None

### CODE REVIEW

File: .github/workflows/containerd.yml [hunk 1]
```diff
@@ -10,7 +10,24 @@ jobs:
       matrix:
         os: [ubuntu-18.04, windows-2019]
         version: [master, v1.4.0-beta.2]
-    name: ${{matrix.version}} / ${{ matrix.os }}
+        runtime: [io.containerd.runtime.v1.linux, io.containerd.runc.v1, io.containerd.runc.v2, containerd-shim-runhcs-v1]
+        runc: [runc, crun]
+        exclude:
+          - runtime: io.containerd.runc.v1
```
Comment (mikebrow) [inline .github/workflows/containerd.yml:16]: feels like a text table would be nice here :-)

File: .github/workflows/containerd.yml [hunk 2]
```diff
@@ -71,6 +99,14 @@ jobs:
           make install
         working-directory: ${{ github.workspace }}/src/github.com/containerd/containerd
 
+      - name: Install ${{matrix.runc}} runtime engine on Linux
+        if: startsWith(matrix.os, 'ubuntu')
+        env:
+          RUNC_FLAVOR: ${{matrix.runc}}
+        run: |
+          gosu root script/setup/install-runc
```
Comment (mikebrow) [inline .github/workflows/containerd.yml:107]: Is `gosu root` required here?  or can you just call script/setup/install-runc and it works successfully? I see containerd ci has it, just wondering if it is no longer required.

**Comment by mikebrow** [comment]
oops, missed the review window :-)

**Comment by saschagrunert** [comment]
> oops, missed the review window :-)

Let’s follow-up on this, the next time I only approve to let you review too. Sorry for that!

**Comment by hickeyma** [comment]
Thanks for merging @saschagrunert.

Thanks for the review @mikebrow. I will follow up on your comments in a follow on PR.

File: .github/workflows/containerd.yml (hunk 1 above)
Comment (hickeyma) [reply .github/workflows/containerd.yml:16]: Added

File: .github/workflows/containerd.yml (hunk 2 above)
Comment (hickeyma) [reply .github/workflows/containerd.yml:107]: Good point. Checked and not needed, so just calling the script direct.

**Comment by hickeyma** [comment]
Opened #648 to handle @mikebrow https://github.com/kubernetes-sigs/cri-tools/pull/647#pullrequestreview-461887086

**Pull request merged**